flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/common/utils/AvroToBigQuerySchemaTransform.java

```diff
@@ -0,0 +1,472 @@
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.BOOLEAN, StandardSQLTypeName.BOOL);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.ENUM, StandardSQLTypeName.STRING);
+        AVRO_TYPES_TO_BQ_TYPES.put(Schema.Type.BYTES, StandardSQLTypeName.BYTES);
+
+        /*
+         * Map Logical Avro Schema Type to StandardSQLTypeName Type, which converts
+         * AvroSchema Logical Type to StandardSQLTypeName.
+         * LOGICAL_AVRO_TYPES_TO_BQ_TYPES: Map containing mapping from Logical Avro Schema Type to StandardSQLTypeName.
+         */
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES = new HashMap<>();
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(LogicalTypes.date().getName(), StandardSQLTypeName.DATE);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timestampMicros().getName(), StandardSQLTypeName.TIMESTAMP);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timestampMillis().getName(), StandardSQLTypeName.TIMESTAMP);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.uuid().getName(), StandardSQLTypeName.STRING);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timeMillis().getName(), StandardSQLTypeName.TIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timeMicros().getName(), StandardSQLTypeName.TIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.localTimestampMillis().getName(), StandardSQLTypeName.DATETIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.localTimestampMicros().getName(), StandardSQLTypeName.DATETIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put("geography_wkt", StandardSQLTypeName.GEOGRAPHY);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put("Json", StandardSQLTypeName.JSON);
+    }
+
+    /**
+     * Converts an Avro schema to a BigQuery schema.
+     *
+     * <p>This method transforms an Avro {@link Schema} into a BigQuery {@link
+     * com.google.cloud.bigquery.Schema}. It iterates through the fields of the Avro schema,
+     * converts each field to its BigQuery equivalent, and constructs a BigQuery schema with the
+     * resulting fields.
+     *
+     * <p>For each Avro field, the method extracts the field's name and documentation (if
+     * available), and uses this information to build the corresponding BigQuery field.
+     *
+     * <p>The Avro schema must be of type Avro RECORD
+     *
+     * @param avroSchema The Avro schema to convert.
+     * @return The converted BigQuery {@link com.google.cloud.bigquery.Schema}.
+     */
+    public static com.google.cloud.bigquery.Schema getBigQuerySchema(Schema avroSchema) {
+        if (avroSchema == null) {
+            throw new IllegalArgumentException(
+                    "Could not convert avro schema of expected output to BigQuery table schema. The avro schema of the expected output record cannot be null");
```

Comment:
Consider line wrapping beyond 100 characters: go/java-style#s4.4-column-limit

Reply:
This can be applied to entire PR

Reply:
Done.